src/components/two-factor-authentication/HardwareTokenWorkflow.vue

```diff
@@ -0,0 +1,252 @@
+    if (isFormDirty()) {
+      showUnsavedChangesDialog.value = true;
+      blockedNext = next;
+    } else {
+      next();
+    }
+  });
+
+  function handleConfirmUnsavedChanges(): void {
+    blockedNext();
+  }
+
+  onMounted(async () => {
+    /* listen for browser changes and prevent them when form is dirty */
+    window.addEventListener('beforeunload', preventNavigation);
+  });
+
+  onUnmounted(() => {
+    window.removeEventListener('beforeunload', preventNavigation);
+  });
+
+  async function assignHardwareToken(): Promise<void> {
+    if (!props.person.person.referrer) return;
+    try {
+      const assignHardwareTokenBodyParams: AssignHardwareTokenBodyParams = {
+        serial: selectedSeriennummer.value,
+        otp: selectedOtp.value,
+        referrer: props.person.person.referrer,
+        userId: props.person.person.id,
+      };
+      await twoFactoreAuthentificationStore.assignHardwareToken(assignHardwareTokenBodyParams);
+      dialogText.value = t('admin.person.twoFactorAuthentication.hardwareTokenSetUpSuccess');
+    } catch (error) {
+      emits('updateHeader', t('admin.person.twoFactorAuthentication.hardwareTokenSetUpFailure'));
+      handleApiError(error);
+    } finally {
+      hardwareTokenIsAssigned.value = true;
+    }
+  }
+
+  const onSubmit: (e?: Event | undefined) => Promise<Promise<void> | undefined> = handleSubmit(async () => {
+    assignHardwareToken();
+    resetForm();
+  });
+</script>
+
+<template v-slot:activator="{ props }">
+  <v-container v-if="!hardwareTokenIsAssigned">
+    <FormWrapper
+      :confirmUnsavedChangesAction="handleConfirmUnsavedChanges"
+      :createButtonLabel="$t('admin.person.twoFactorAuthentication.setUp')"
+      :discardButtonLabel="$t('cancel')"
+      id="schule-creation-form"
+      :onDiscard="cancelCheck"
+      @onShowDialogChange="(value?: boolean) => (showUnsavedChangesDialog = value ?? false)"
+      :centerButtons="true"
+      :onSubmit="onSubmit"
+      :showUnsavedChangesDialog="showUnsavedChangesDialog"
+    >
+      <FormRow
+        :errorLabel="''"
+        labelForId="serial-input"
+        :isRequired="true"
+        :label="t('admin.person.twoFactorAuthentication.serial')"
+      >
+        <v-text-field
+          clearable
+          data-testid="hardware-token-input"
+          density="compact"
+          :disabled="false"
+          id="hardwareToken-input"
+          :placeholder="t('admin.person.twoFactorAuthentication.serial')"
+          ref="hardwareToken-input"
+          required="true"
+          variant="outlined"
+          v-model="selectedSeriennummer"
+          v-bind="selectedSeriennummerProps"
+        ></v-text-field>
+      </FormRow>
+      <FormRow
+        :errorLabel="''"
+        labelForId="otp-input"
+        :isRequired="true"
+        :label="t('admin.person.twoFactorAuthentication.currentOtp')"
+      >
+        <v-text-field
+          clearable
+          data-testid="hardwareToken-input"
```

Comment:
```suggestion
          data-testid="hardware-token-input"
```
ids und refs der inputs haben teilweise noch andere schreibweise, bitte alle felder prüfen.